v2/AdminCore.Services/EventService.cs

```diff
@@ -302,5 +333,38 @@ private bool IsHolidayDatesAlreadyBooked(int employeeId, EventDateDto eventDates
 
       return true;
     }
+
+    private bool DoesEmployeeHaveEnoughHolidays(int employeeId, Event newEvent)
+    {
+      return GetHolidayStatsForUser(employeeId).AvailableHolidays >= GetDaysInEvent(newEvent.EventDates);
+    }
+
+    private double GetDaysInEvent(ICollection<EventDate> newEventEventDates)
+    {
+      double totalDays = 0;
+      foreach (var eventDate in newEventEventDates)
+      {
+        if (IsSameDay(eventDate) && eventDate.IsHalfDay)
```

Comment:
Extract 347 - 354 into own method. Something like this
```suggestion
        totalDays += GetDaysFromEventDate(eventDate)
```